tests/test_esm_runscripts/test_filedicts.py

```diff
@@ -159,6 +159,60 @@ def test_filedicts_basics(fs):
     assert sim_file.locations["computer"] == Path("/work/ollie/pool/ECHAM/T63")
 
 
+def test_path_type(simulation_file, fs):
+    # check for incompatible type
+    with pytest.raises(TypeError):
+        output = simulation_file._path_type(12312312)
+
+    # check for non-existent directory / file
+    path = "/this/path/does/not/exist" 
+    output = simulation_file._path_type(path)
+    assert output == None
+
+    # check for link
+    mylink = "/tmp/mylink"
+    fs.create_symlink(mylink, simulation_file["absolute_path_in_computer"])
+    output = simulation_file._path_type(mylink)
+    assert output == "link"
+
+    # check for file
+    output = simulation_file._path_type(simulation_file["absolute_path_in_computer"])
+    assert output == "file"
+
+    # check for directory
+    output = simulation_file._path_type(simulation_file.locations["work"])
+    assert output == "dir"
+
+
+def test_check_source_and_targets(simulation_file, fs):
+    # successful return
+    path_str = "/home/ollie/dural/test_dir"
+    fs.create_dir(path_str)
+    source_path = simulation_file.locations["computer"] 
+    target_path = Path(f"{path_str}/file.txt")
+    output = simulation_file._check_source_and_target(source_path, target_path)
+    assert output == True
+
+    # check incompatible types for file paths
+    # TODO: when `_convert_to_path` was removed, these tests fail
+    with pytest.raises(TypeError):
+        simulation_file._check_source_and_target(Path("/usr/bin"), {"foo": 1})
+    with pytest.raises(TypeError):
+        simulation_file._check_source_and_target(1234, 3.1415)
+
+    # source does not exist
+    source_path = Path("/this/does/not/exist")
+    target_path = simulation_file.locations["work"]
+    with pytest.raises(FileNotFoundError):
+        simulation_file._check_source_and_target(source_path, target_path)
+
+    # target does not exist
+    source_path = simulation_file.locations["computer"] 
+    target_path = Path("/this/does/not/exist")
+    with pytest.raises(FileNotFoundError):
+        simulation_file._check_source_and_target(source_path, target_path)
```

Comment:
I think this can also be devided in different tests. One can turn the comments into test descriptions once the tests are separated.

Reply:
Good idea, I will do it.

Reply:
Already implemented and pushed.